sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,244 @@
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in settings like WebBTC (see WBIP-002) and other more Ethereum inspired domains.
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+### Method Independent
+
+#### Common Definitions
+
+The following definitions can be used in multiple methods (mainly for transfer and transaction methods).
+
+`params`
+
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` (anything parseable by the BigInt constructor)
+- `nonce?`: `number | string` (anything parseable by the BigInt constructor)
+- `attachment?`: `string` hex-encoded
+- `anchoreMode?`: `'onChainOnly' | 'offChainOnly' | 'any'`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `sponsored?`: `boolean`, defaults to `false`
+- `address?`: `string` address, Stacks c32-encoded (sender or signer address), defaults to wallets current address
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string` hex-encoded
+
+### `stx_transferStx`
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` (anything parseable by the BigInt constructor)
+- `memo?`: `string`, defaults to `''`
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### `stx_transferFt`
+
+`todo: haven't existed yet, should we add them?`
+
+### `stx_transferNft`
+
+`todo: haven't existed yet, should we add them?`
+
+### `stx_contractCall`
+
+`params`
+
+- `contractAddress`: `string` address, Stacks c32-encoded
+- `contractName`: `string`
+- `functionName`: `string`
+- `functionArgs`: `ClarityValue[]`, defaults to `[]`
+
+`where`
+
+- `ClarityValue`: `string` hex-encoded
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### `stx_contractDeploy`
+
+`params`
+
+- `contractName`: `string`
+- `codeBody`: `string` Clarity contract code
+- `clarityVersion?`: `number`
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### `stx_signTransaction`
+
+`params`
+
+- `transaction`: `string` hex-encoded
+
+`result`
+
+- `transaction`: `string` hex-encoded (signed)
+
+### `stx_signMessage`
+
+`params`
+
+- `message`: `string`
+
+`result`
+
+- `signature`: `string` hex-encoded
+- `publicKey`: `string` hex-encoded
+
+### `stx_signStructuredMessage`
+
+`params`
+
+- `message`: `string` Clarity value, hex-encoded
+- `domain?`: `string` hex-encoded (defined by SIP-018)
```

Comment:
What's making the `domain` be optional? Shouldn't `stx_signMessage` be used when there's no domain? Alternatively, would it make sense to have a single `stx_signMessage` method with an optional `domain?` param?